### PR TITLE
fix: remove duplicate summaries

### PR DESCRIPTION
**Pull request by IgboPharaoh**
Removes duplicate summaries from summary list from a specific batch

### CODE REVIEW

**Comment by adamjonas** [comment]
still seeing `MATT: [demo] Optimistic execution of arbitrary programs` listed twice. Once as a single post and once as a combined summary - is this supposed to remove that?

**Comment by IgboPharaoh** [comment]
> still seeing `MATT: [demo] Optimistic execution of arbitrary programs` listed twice. Once as a single post and once as a combined summary - is this supposed to remove that?

This is actually supposed to remove that. The `MATT: [demo] Optimistic execution of arbitrary programs` which you see as a single post is coming from another month. This happens because we're fetching, and cleaning each batch according to months.

**Comment by adamjonas** [comment]
So can anything be done? What is the downside of loading and fetching bigger batches of posts?

**Comment by IgboPharaoh** [comment]
The only downside of fetching bigger posts is longer fetching time.

**Comment by adamjonas** [comment]
How about we try to load the landing page with 3 months and fetch 2 or 3 months depending on load time. Any downsides to that approach?

**Comment by IgboPharaoh** [comment]
No downsides to that.

**Comment by IgboPharaoh** [comment]
The major reason why we are seeing summaries from a previous month is, that's the time/month when a conversation on a particular topic started. So fetching in batches of 3 months is working on an assumption that in the space of 3 months, a conversation would most likely be resolved and a combined summary generated. 

I think this further eliminates to a greater percentage the possibility  of seeing a prior conversation

**Comment by adamjonas** [comment]
Let's give it a try

**Comment by adamjonas** [comment]
ACK on functionality

File: src/helpers/utils.ts [hunk 1]
```diff
@@ -159,3 +156,41 @@ export const createMonthsFromKeys = (startYear: number, endYear: number) => {
...
+  }
+
+  const singleSummaries = Object.keys(cleanedGroups)
+    .filter((post) => !post.toLowerCase().startsWith("combined summary"))
+    .map((x) => x.toLowerCase());
+
+  const combinedSummaries = Object.keys(cleanedGroups)
+    .filter((post) => post.toLowerCase().startsWith("combined summary"))
+    .map((x) => x.toLowerCase());
+
```
Comment (Emmanuel-Develops) [inline src/helpers/utils.ts:177]: you might want to add a comment here, as it gets tricky of what exactly the code does until detailed inspect

**Comment by Emmanuel-Develops** [comment]
looks good 👍
merging this!

**Pull request merged**